api/src/main/java/org/pmiops/workbench/api/OfflineUserController.java

```diff
@@ -42,7 +44,10 @@ public ResponseEntity<Void> sendAccessExpirationEmails() {
   }
 
   public ResponseEntity<Void> checkInitialCreditsExpiration() {
-    taskQueueService.groupAndPushCheckInitialCreditExpirationTasks(userService.getAllUserIds());
+    taskQueueService.groupAndPushCheckInitialCreditExpirationTasks(
+        userService.getAllUsersWithActiveInitialCredits().stream()
```

Comment:
This only check the expiration records of users who have expiration records and those who have not already been cleaned up.